cmd/build.go

```diff
@@ -0,0 +1,203 @@
+	flags.StringVar(&job, "job", job, "job to be executed")
+	flags.BoolVar(&skipCheckout, "skip-checkout", skipCheckout, "use local path as-is")
+	flags.IntVar(&nodeTotal, "node-total", nodeTotal, "total number of parallel nodes")
+
+	return command
+}
+
+var picardRepo = "circleci/picard"
+var circleCiDir = path.Join(settings.UserHomeDir(), ".circleci")
+var currentDigestFile = path.Join(circleCiDir, "current_picard_digest")
+var latestDigestFile = path.Join(circleCiDir, "latest_picard_digest")
+
+func getDigest(file string) string {
+
+	if _, err := os.Stat(file); !os.IsNotExist(err) {
+		digest, err := ioutil.ReadFile(file)
+
+		if err != nil {
+			Logger.Error("Could not load digest file", err)
+		} else {
+			return strings.TrimSpace(string(digest))
+		}
+	}
+
+	return "" // Unknown digest
+}
+
+func newPullLatestImageCommand() *exec.Cmd {
+	return exec.Command("docker", "pull", picardRepo)
+}
+
+func getLatestImageSha() (string, error) {
+
+	cmd := newPullLatestImageCommand()
+	Logger.Info("Pulling latest build image")
+	bytes, err := cmd.CombinedOutput()
+
+	if err != nil {
+		return "", errors.Wrap(err, "failed to pull latest image")
+	}
+
+	output := string(bytes)
+
+	sha256 := regexp.MustCompile("(?m)sha256.*$")
+
+	//latest_image=$(docker pull picardRepo | grep -o "sha256.*$")
+	//echo $latest_image > $LATEST_DIGEST_FILE
+	latest := sha256.FindString(output)
+
+	if latest == "" {
+		return latest, errors.New("Failed to find latest image")
+	}
+
+	return latest, nil
+}
+
+func picardImage() (string, error) {
+	currentDigest := getDigest(currentDigestFile)
+
+	if currentDigest == "" {
+		// Receiving latest image of picard in case of there's no current digest stored
+		Logger.Info("Downloading latest CircleCI build agent...")
+
+		var err error
+
+		currentDigest, err = getLatestImageSha()
+
+		// TODO - write the digest to a file so that we can
+		// use it again.
+
+		if err != nil {
+			return "", err
+		}
+
+	} else {
+		// Otherwise pulling latest image in background
+		// TODO - this should write to a file to record
+		// the fact that we have the latest image.
+		newPullLatestImageCommand().Start()
+
+	}
+
+	Logger.Infof("Docker image digest: %s", currentDigest)
+
+	return fmt.Sprintf("%s@%s", picardRepo, currentDigest), nil
+}
+
+func streamOutout(stream io.Reader) {
+	scanner := bufio.NewScanner(stream)
+	go func() {
+		for scanner.Scan() {
+			Logger.Info(scanner.Text())
+		}
+	}()
+}
+
+func runBuild(cmd *cobra.Command, args []string) error {
+
+	ctx := context.Background()
+
+	pwd, err := os.Getwd()
+
+	if err != nil {
+		return errors.Wrap(err, "Could not find pwd")
+	}
+
+	image, err := picardImage()
+
+	if err != nil {
+		return errors.Wrap(err, "Could not find picard image")
+	}
+
+	// TODO: marc:
+	// I can't find a way to pass `-it` and have carriage return
+	// characters work correctly.
+	arguments := []string{"run", "--rm",
+		"-e", fmt.Sprintf("DOCKER_API_VERSION=%s", dockerAPIVersion),
```

Comment:
This doesn't need to be a flag. Use `-e DOCKER_API_VERSION` and it will use the value from `cmd.Env`, which defaults to `os.Environ()` which is exactly what you want I think.